HHAuto.user.js

```diff
@@ -6886,6 +6899,38 @@ var CrushThemFights=function()
         }
     }
 
+    if (Storage().HHAuto_Setting_useX10Fights === "true"
+        && Storage().HHAuto_Setting_minShardsX10
+        && Number.isInteger(Number(Storage().HHAuto_Setting_minShardsX10))
+        && remainingShards >= Number(Storage().HHAuto_Setting_minShardsX10)
+        && (battleButtonX10Price === 0 || getSetHeroInfos('hard_currency')>=battleButtonX10Price+Number(Storage().HHAuto_Setting_kobanBank))
+        && Number( getSetHeroInfos('fight.amount')) >= 10
+        && (Number(getSetHeroInfos('fight.amount')) >= (Number(Storage().HHAuto_Setting_autoTrollThreshold) + 10)
+            || bypassThreshold
+           )
+       )
+    {
+        logHHAuto("Going to crush 50 times: "+Trollz[Number(TTF)]+' for '+battleButtonX50Price+' kobans.');
+
+        hero.infos.hc_confirm = true;
+        // We have the power.
+        is_cheat_click=function(e) {
+            return false;
+        };
+        battleButtonX50.click();
+        hero.infos.hc_confirm = hcConfirmValue;
+        logHHAuto("Crushed 50 times: "+Trollz[Number(TTF)]+' for '+battleButtonX50Price+' kobans.');
+        setTimeout(function(){gotoPage('home');},randomInterval(300,500));//gotoPage('home');
+        return;
+    }
+    else
+    {
+        if (Storage().HHAuto_Setting_useX50Fights === "true")
+        {
+            logHHAuto('Unable to use x50 for '+battleButtonX50Price+' kobans,fights : '+getSetHeroInfos('fight.amount')+'/50, remaining shards : '+remainingShards+'/'+Storage().HHAuto_Setting_minShardsX50+', kobans : '+getSetHeroInfos('hard_currency')+'/'+Number(Storage().HHAuto_Setting_kobanBank));
+        }
+    }
+
```

Comment:
@YotoTheOne these ones I don't understand, seems a duplicate mix for x10 and x50

Reply:
Right, that must have happened when I merged your main with my better-UI.
Solved !

Reply:
@YotoTheOne, yes merge in github is not that easy

Reply:
@Roukys : sorry for that, I wanted to simplify the work for you since you're kind enough to accept my upgrades ;)
That happened because the function x10 x50 is brand new and I didn't have time to get to know it. 
I think that function was not there when I started the pull request. 
Anyway I understand better how it works now, I'll be able to help more without complicating your work. 
UI is mostly done, maybe I'll do a correction here and there but nothing with too much code, and I'll do little at a time. 

Do you know a better tool than GitHub ? I find it pretty useful with the ability to compare code and revert changes.